### PR TITLE
spatialite-gui: update to 2.1.0

### PR DESCRIPTION
**Pull request by danielnachun**
- [ ] Have you followed the [guidelines for contributing](https://github.com/Homebrew/homebrew-core/blob/HEAD/CONTRIBUTING.md)?
- [ ] Have you ensured that your commits follow the [commit style guide](https://docs.brew.sh/Formula-Cookbook#commit)?
- [ ] Have you checked that there aren't other open [pull requests](https://github.com/Homebrew/homebrew-core/pulls) for the same formula update/change?
- [ ] Have you built your formula locally with `brew install --build-from-source <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Is your test running fine `brew test <formula>`, where `<formula>` is the name of the formula you're submitting?
- [ ] Does your build pass `brew audit --strict <formula>` (after doing `brew install --build-from-source <formula>`)? If this is a new formula, does it pass `brew audit --new <formula>`?

-----

### CODE REVIEW

**Comment by danielnachun** [comment]
As discussed in https://github.com/Homebrew/homebrew-core/pull/94807#issuecomment-1036060015, despite being tagged as a beta, this version of `spatialite-gui` is being packaged as a stable release by Debian and Fedora.  We need to upgrade to it because it provides PROJ 8 support to `spatialite-gui`, which is needed for  https://github.com/Homebrew/homebrew-core/pull/94807 to work.

In addition to upgrading `spatialite-gui`, we can also deprecate `librasterlite` and `libgaiagraphics`, both of which have been deprecated upstream because the have been replaced by `librasterlite2`.

File: Formula/spatialite-gui.rb [hunk 1]
```diff
@@ -24,20 +23,29 @@ class SpatialiteGui < Formula
...
+  depends_on "libtiff"
+  depends_on "libxlsxwriter"
+  depends_on "lz4"
+  depends_on "minizip"
+  depends_on "openjpeg"
   depends_on "proj@7"
   depends_on "sqlite"
+  depends_on "virtualpg"
+  depends_on "webp"
   depends_on "wxwidgets@3.0"
```
Comment (carlocab) [inline Formula/spatialite-gui.rb:31]: Since we're updating this we should check whether it supports the newer `wxwidgets`.

**Review by carlocab — CHANGES_REQUESTED** [review]
Let's do the deprecations in a separate PR to avoid having to rebuild these formulae and discarding bottles unnecessarily.

File: Formula/spatialite-gui.rb [hunk 2]
```diff
@@ -1,10 +1,9 @@
 class SpatialiteGui < Formula
   desc "GUI tool supporting SpatiaLite"
   homepage "https://www.gaia-gis.it/fossil/spatialite_gui/index"
-  url "https://www.gaia-gis.it/gaia-sins/spatialite-gui-sources/spatialite_gui-1.7.1.tar.gz"
-  sha256 "cb9cb1ede7f83a5fc5f52c83437e556ab9cb54d6ace3c545d31b317fd36f05e4"
+  url "https://www.gaia-gis.it/gaia-sins/spatialite-gui-sources/spatialite_gui-2.1.0-beta1.tar.gz"
+  sha256 "ba48d96df18cebc3ff23f69797207ae1582cce62f4596b69bae300ca3c23db33"
```
Comment (carlocab) [inline Formula/spatialite-gui.rb:5]: We could probably also do this version bump at the same time as we migrate `gdal`'s dep tree to `proj`.

File: Formula/spatialite-gui.rb (hunk 1 above)
Comment (danielnachun) [reply Formula/spatialite-gui.rb:31]: Unfortunately I tried this and it doesn't work.  This is even mentioned in a comment in the AUR page for `spatialite-gui`: https://aur.archlinux.org/packages/spatialite-gui.

File: Formula/spatialite-gui.rb (hunk 2 above)
Comment (danielnachun) [reply Formula/spatialite-gui.rb:5]: In that case, I think I should close this PR and do this in https://github.com/Homebrew/homebrew-core/pull/94807 instead.

**Comment by danielnachun** [comment]
Based on the above comment, rather than handling this in a separate PR, I will update `spatialite-gui` in https://github.com/Homebrew/homebrew-core/pull/94807 instead.

**Pull request closed without merging**